src/manifolds/Hyperbolic.jl

```diff
@@ -1,116 +1,221 @@
+
 # Constructor
 
     Hyperbolic(n)
 
-Generate the $ℍ^{n} ⊂ ℝ^{n+1}$
+Generate the Hyperbolic manifold of dimension `n`.
 """
 struct Hyperbolic{N} <: AbstractEmbeddedManifold{ℝ,DefaultIsometricEmbeddingType} end
 
 Hyperbolic(n::Int) = Hyperbolic{n}()
 
+@doc raw"""
+    HyperboloidPoint <: MPoint
+
+In the Hyperboloid model of the [`Hyperbolic`](@ref) $\mathcal H^n$ points are represented
+as vectors in $ℝ^{n+1}$ with [`MinkowskiMetric`](@ref) equal to $-1$.
+
+This representation is the default, i.e. vectors are assumed to have this repesentation.
 """
-    check_manifold_point(M::Hyperbolic, p; kwargs...)
+struct HyperboloidPoint{TValue<:AbstractVector} <: MPoint
+    value::TValue
+end
 
-Check whether `p` is a valid point on the [`Hyperbolic`](@ref) `M`, i.e. is a `p` vector with
-inner product in the embedding of -1, see [`MinkowskiMetric`](@ref).
-The tolerance for the last test can be set using the `kwargs...`.
+@doc raw"""
+    HyperboloidTVector <: TVector
+
+In the Hyperboloid model of the [`Hyperbolic`](@ref) $\mathcal H^n$ tangent vctors are represented
+as vectors in $ℝ^{n+1}$ with [`MinkowskiMetric`](@ref) $⟨p,X⟩_{\mathrm{M}}=0$ to their base
+point $p$.
+
+This representation is the default, i.e. vectors are assumed to have this repesentation.
 """
-function check_manifold_point(M::Hyperbolic, p; kwargs...)
-    mpv =
-        invoke(check_manifold_point, Tuple{supertype(typeof(M)),typeof(p)}, M, p; kwargs...)
-    mpv === nothing || return mpv
-    if !isapprox(minkowski_metric(p, p), -1.0; kwargs...)
-        return DomainError(
-            minkowski_metric(p, p),
-            "The point $(p) does not lie on $(M) since its Minkowski inner product is not -1.",
-        )
+struct HyperboloidTVector{TValue<:AbstractVector} <: MPoint
+    value::TValue
+end
+
+@doc raw"""
+    PoincareBallPoint <: MPoint
+
+A point on the [`Hyperbolic`](@ref) manifold $\mathcal H^n$ can be represented as a vector of norm
+less than one in $\mathbb R^n$.
+"""
+struct PoincareBallPoint{TValue<:AbstractVector} <: MPoint
+    value::TValue
+end
+
+@doc raw"""
+    PoincareBallTVector <: TVector
+
+In the Poincaré ball model of the [`Hyperbolic`](@ref) $\mathcal H^n$ tangent vctors are represented
+as vectors in $ℝ^{n}$.
+"""
+struct PoincareBallTVector{TValue<:AbstractVector} <: MPoint
+    value::TValue
+end
+
+@doc raw"""
+    PoincareHalfSpacePoint <: MPoint
+
+A point on the [`Hyperbolic`](@ref) manifold $\mathcal H^n$ can be represented as a vector in the
+half plane, i.e. $x ∈ ℝ^n$ with $x_d > 0$.
+"""
+struct PoincareHalfSpacePoint{TValue<:AbstractVector} <: MPoint
+    value::TValue
+end
+
+@doc raw"""
+    PoincareHalfPlaneTVector <: TVector
+
+In the Poincaré half plane model of the [`Hyperbolic`](@ref) $\mathcal H^n$ tangent vctors are
+represented as vectors in $ℝ^{n}$.
+"""
+struct PoincareHalfSpaceTVector{TValue<:AbstractVector} <: MPoint
```

Comment:
```suggestion
struct PoincareHalfSpaceTVector{TValue<:AbstractVector} <: TVector
```

Reply:
Oops. Thanks for noticing, luckily our implementations are not that dependent, so everything still works.